### PR TITLE
CDAP-7984 Correct descriptions in cdap-defaults.xml

### PR DESCRIPTION
**Pull request by awholegunch**
Changes to cdap-default.xml descriptions. Re-organizes the appendix page to have the notes at the top, and adds a note about the Kafka service properties.
Fixes errors in the building of the rst file; fixes issues with the use of
HTML entity-equivalents in the descriptions, and adds a method to enclose
phrases that start with "${" or "<" with literals. Updates the MD5 hash.

In particular, it fixes the description of `messaging.system.topics`.

Building as a Quick Build: http://builds.cask.co/browse/CDAP-DQB250-1

Page of interest: http://builds.cask.co/artifact/CDAP-DQB250/shared/build-1/Docs-HTML/html/admin-manual/appendices/cdap-site.html

Fix for: https://issues.cask.co/browse/CDAP-7984
(`kafka.server.log.retention.hours` and `log.retention.duration.days`)

### CODE REVIEW

File: cdap-common/src/main/resources/cdap-default.xml [hunk 1]
```diff
@@ -1546,10 +1549,10 @@
     <name>metrics.messaging.topic.num</name>
     <value>10</value>
     <description>
-      Number of topics for metrics in messaging. This number controls how many topics in messaging
-      which metrics are sent to, and the number of threads fetching from messaging service
-      and processing metrics in parallel. With the default value 10, 10 topics will be created for metrics
-      with names: ${metrics.topic.prefix}0, ${metrics.topic.prefix}1, ... ${metrics.topic.prefix}9
+      Number of topics for metrics in messaging. This property controls how many topics in messaging
+      that metrics are sent to, and the number of threads used to fetch from the messaging service
```
Comment (anew) [inline cdap-common/src/main/resources/cdap-default.xml]: this sentence is confusing...

**Comment by anew** [comment]
Only one comment, other than that LGTM

File: cdap-common/src/main/resources/cdap-default.xml (hunk 1 above)
Comment (awholegunch) [reply cdap-common/src/main/resources/cdap-default.xml]: OK...let me try again.

**Comment by anew** [comment]
lgtm
